core/src/main/java/com/scalar/db/exception/transaction/AbortException.java

```diff
@@ -13,4 +15,17 @@ public AbortException(String message, String transactionId) {
   public AbortException(String message, Throwable cause, String transactionId) {
     super(message, cause, transactionId);
   }
+
+  public AbortException(
+      String message, @Nullable String transactionId, boolean authenticationError) {
+    super(message, transactionId, authenticationError, false, false, null);
+  }
+
+  public AbortException(
+      String message,
+      Throwable cause,
+      @Nullable String transactionId,
+      boolean authenticationError) {
```

Comment:
Instead of adding an argument only for authentication, it might be better to define an error code and add it here or something?
Since we don't define error code yet, I understand it's hard to do so now...

Reply:
Yeah, at the moment, it's hard to do so since we need to define error code for all errors for that. Anyway, I think we can add error codes later on.